ash1eygrace/ai-content
# Generators: Dynamically render generators by passing data through React routes

**Issue this solves:** 

[Refactor: Generators by Passing Data to Components via Routes ](https://github.com/ash1eygrace/ai-content/issues/20) #20 

**Changes in this PR:** 

In this pull request, I have refactored our codebase to simplify and improve the overall maintainability of our application. Previously, I had separate components for BlogIdeas.js, BlogIntro.js, CompanyBio.js, LinkedInJobDescription.js, ProductDescription.js, and Tldr.js, which shared similar functionality and structure.

To make our code more maintainable and efficient, I have combined these components into a single component called GeneratorComponent.js. This new component is designed to handle the functionality of all the previous individual components.

The data for these components have been centralized in the data/GeneratorList.js file, which is now being passed to the GeneratorComponent.js through React Router routes. This approach allows us to easily manage and update the data for each generator while keeping our component code clean and DRY (Don't Repeat Yourself).

Changes:
- Updated the data/GeneratorList.js file to include all the necessary data for the generator components.
- Refactored the React Router routes in App.js to pass the data from data/GeneratorList.js to the GeneratorComponent.js component.
- Removed individual components: BlogIdeas.js, BlogIntro.js, CompanyBio.js, LinkedInJobDescription.js, ProductDescription.js, and Tldr.js.
- Created a new GeneratorComponent.js component that handles the functionality of the previously separated components.

These changes will make our application more maintainable, easier to extend, and provide a more consistent user experience.

**Testing Instructions:** 

1. Follow the [installation guide](https://github.com/ash1eygrace/ai-content#installation). 
2. Click on Generators in the menu
3. Test opening all the apps by clicking on each of their buttons. E.g. in the Blog Post Ideas card, click the button titled Blog Post Ideas.

Expected: The Blog Post Ideas generator page should open with the Blog Post Idea content form and card to generate a response. 

4. Test the rest of the apps to make sure they all load with their respective content passed from the data/GeneratorList.js file
5. Test the generators by adding your query into the form to ensure it responds with the correct contextual response.
